### PR TITLE
Add CNAME support and gotchas

### LINKED ISSUE #9680: Certificate renewal and generation fails for wildcard domain using gandiv5 DNS challenge
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

Upgrading from docker image for Traefik 2.9.1 to 2.9.4 and subsequent (bringing lego 4.9.0+).
I am unable to process a cert renewal (wildcard gandiv5 dns challenge) with the following error :

`time="2023-01-28T17:13:58+01:00" level=error msg="Error renewing certificate from LE: {subdomain.domain.name [*.subdomain.domain.name]}" providerName=letsencrypt.acme error="error: one or more domains had a problem:\n[*.subdomain.domain.name] [*.subdomain.domain.name] acme: error presenting token: gandiv5: unexpected authZone domain.name. for fqdn domain.name.\n[subdomain.domain.name] [subdomain.domain.name] acme: error presenting token: gandiv5: unexpected authZone domain.name. for fqdn domain.name.\n" ACME CA="https://acme-v02.api.letsencrypt.org/directory"`

### What did you see instead?

The issue seems related to the packaged lego component from version 4.9.0 brought with Traefik 2.9.4 and subsequent.

Since [v4.9.0](https://github.com/go-acme/lego/releases/tag/v4.9.0) lego follows CNAME as expected by the ACME, see https://letsencrypt.org/2019/10/09/onboarding-your-customers-with-lets-encrypt-and-acme.html#the-advantages-of-a-cname

In order to allow certificate generation or renewal in such a case, one must add the following to the Traefik docker definition file : 

`      - LEGO_DISABLE_CNAME_SUPPORT=true`

### What version of Traefik are you using?

2.9.6

### What is your environment & configuration?

```yaml

    environment:
      - GANDIV5_API_KEY=APIKEYGOESHERE
    
    labels:
      - "traefik.http.routers.traefik-secure.tls.certresolver=letsencrypt"
      - "traefik.http.routers.traefik-secure.tls.domains[0].main=subdomain.domain.name"
      - "traefik.http.routers.traefik-secure.tls.domains[0].sans=*.subdomain.domain.name"
```

Add more configuration information here.
traefik.yml

certificatesResolvers:
  letsencrypt:
    acme:
      email: emacron@gmail.com
      storage: /acme.json
      dnsChallenge:
        provider: gandiv5
        resolvers:
          - "1.1.1.1:53"
          - "8.8.8.8:53"

### If applicable, please paste the log output in DEBUG level

_No response_

### PR DESCRIPTION
**Pull request by mpl**
### What does this PR do?

This PR documents CNAME support in DNS challenge for acme.

### Motivation

Users are getting some errors in some very specific cases, related to CNAME support.

### More

~- [ ] Added/updated tests~
- [x] Added/updated documentation

### Additional Notes

Fixes #9680

### CODE REVIEW

**Comment by mpl** [comment]
@ldez PTAL
@svx FYI

**Review by svx — APPROVED** [review]

File: docs/content/getting-started/faq.md [hunk 1]
```diff
@@ -181,3 +181,19 @@ and the message should help in figuring out the mistake(s) in the configuration,
 
 When using the file provider,
 one easy way to check if the dynamic configuration is well-formed is to validate it with the [JSON Schema of the dynamic configuration](https://json.schemastore.org/traefik-v2-file-provider.json).
+
+## Why does Let's Encrypt wildcard certificate renewal/generation with DNS challenge fail?
+
+If you're trying to renew wildcard certificates, with DNS challenge,
+and you're getting errors such as:
+
+<pre class="highlight"><code class="language-text" style="white-space : pre-wrap !important;">time="2023-01-28T17:13:58+01:00" level=error msg="Error renewing certificate from LE: {subdomain.domain.name [*.subdomain.domain.name]}" providerName=letsencrypt.acme error="error: one or more domains had a problem:\n[*.subdomain.domain.name] [*.subdomain.domain.name] acme: error presenting token: gandiv5: unexpected authZone domain.name. for fqdn domain.name.\n[subdomain.domain.name] [subdomain.domain.name] acme: error presenting token: gandiv5: unexpected authZone domain.name. for fqdn domain.name.\n" ACME CA="https://acme-v02.api.letsencrypt.org/directory"</code></pre>
```
Comment (ldez) [inline docs/content/getting-started/faq.md:190]:
```suggestion

```txt
level=error msg="Error renewing certificate from LE: {example.com [*.example.com]}" providerName=letsencrypt.acme error="error: one or more domains had a problem:\n[example.com] [example.com] acme: error presenting token: gandiv5: unexpected authZone domain.name. for fqdn example.com.\n""
```
```

Comment (mpl) [reply docs/content/getting-started/faq.md:190]: I understand the contents changes (and they're fair enough), but the style change does not achieve the same result.
With the custom style, we get a block of text that is automatically wrapped to the width of the available space, so it is actually readable for the users without having to scroll left and right.

File: docs/content/https/acme.md [hunk 2]
```diff
@@ -283,8 +283,19 @@ Use the `DNS-01` challenge to generate and renew ACME certificates by provisioni
     # ...
     ```
 
-    !!! important
-        A `provider` is mandatory.
+!!! warning "`CNAME` support"
+
+    `CNAME` are supported (and sometimes even [encouraged](https://letsencrypt.org/2019/10/09/onboarding-your-customers-with-lets-encrypt-and-acme.html#the-advantages-of-a-cname)),
+    but there's a few cases where they can be [problematic](../../getting-started/faq/#why-does-lets-encrypt-wildcard-certificate-renewalgeneration-with-dns-challenge-fail).
```
Comment (mpl) [inline docs/content/https/acme.md:289]:
```suggestion
    but there are a few cases where they can be [problematic](../../getting-started/faq/#why-does-lets-encrypt-wildcard-certificate-renewalgeneration-with-dns-challenge-fail).
```

**Review by kevinpollet — APPROVED** [review]
LGTM

**Review by nmengin — APPROVED** [review]
LGTM 📖 👍

**Pull request merged**